services/src/main/java/org/keycloak/vault/PlainTextVaultProvider.java

```diff
@@ -0,0 +1,80 @@
+package org.keycloak.vault;
+
+import org.jboss.logging.Logger;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.EnumSet;
+import java.util.Optional;
+
+/**
+ * A text-based vault provider, which stores each secret in a separate file. The file name needs to match a
+ * vault secret id (or a key for short). A typical vault directory layout looks like this:
+ * <pre>
+ *     ${VAULT}/realma__key1 (contains secret for key 1)
+ *     ${VAULT}/realma__key2 (contains secret for key 2)
+ *     etc...
+ * </pre>
+ * Note, that each key needs is prefixed by realm name. This kind of layout is used by Kubernetes by default
+ * (when mounting a volume into the pod).
+ *
+ * See https://kubernetes.io/docs/concepts/configuration/secret/
+ * See https://github.com/keycloak/keycloak-community/blob/master/design/secure-credentials-store.md#plain-text-file-per-secret-kubernetes--openshift
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class PlainTextVaultProvider implements VaultProvider {
+
+    private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final Path vaultPath;
+    private String realmName;
```

Comment:
```suggestion
    private String final realmName;
```
realmName can be final too

Reply:
Sorry about that... last minute fixes :(